### PR TITLE
Apply pending block details on commit

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

**Comment by NikVolf** [comment]
probably related to #2258

**Comment by gavofyork** [comment]
does indeed look like it's incorrectly failing.

**Comment by rphmeier** [comment]
I don't think we should merge this until a fix is available. It will just make all our other builds fail and won't get us any closer to fixing the issue.

**Comment by NikVolf** [comment]
@rphmeier @gavofyork 
yeah of course it shouldn't be merged, it's just problem formalization
in fact, i have no ideas how to fix that at the moment, so feel free to provide solution

File: ethcore/src/blockchain/blockchain.rs [hunk 1]
```diff
@@ -934,8 +925,10 @@ impl BlockChain {
 				},
 			}
 			let mut write_hashes = self.pending_block_hashes.write();
+			let mut write_details = self.pending_block_details.write();
```
Comment (rphmeier) [inline ethcore/src/blockchain/blockchain.rs]: comment above should now say "all four locks taken"

**Pull request merged**